### PR TITLE
Properties configuration generator

### LINKED ISSUE #253: Add Annotations to Properties for Generating Documentation
We should add annotations to config properties to generate a documentation with description, examples, default values and recommendations automatically, like [wiki/DSF-0.5.0-Configuration-Parameters](https://github.com/highmed/highmed-dsf/wiki/DSF-0.5.0-Configuration-Parameters).

### LINKED ISSUE #307: Create process configuration generator for properties
During the maven build, a markdown based configuration page should be built for properties. This could include:
- property name
- description
- example
- default value
- recommendation

### PR DESCRIPTION
**Pull request by wetret**
Adds an annotation and a generator to automatically create Markdown documentations of properties.

resolves #307 
resolves #253

### CODE REVIEW

File: dsf-tools/dsf-tools-documentation-generator/src/main/java/org/highmed/dsf/tools/generator/DocumentationGenerator.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+					workingPackage, pluginDefinitionClasses.get(0).getName(), exception.getMessage());
+			return Collections.emptyList();
+		}
+	}
+
+	private Optional<String> getProcessName(String bpmnFile)
+	{
+		try
+		{
+			InputStream resource = getClass().getClassLoader().getResource(bpmnFile).openStream();
```
Comment (hhund) [inline dsf-tools/dsf-tools-documentation-generator/src/main/java/org/highmed/dsf/tools/generator/DocumentationGenerator.java:102]: Make sure the InputStream is being closed somewhere. If it does not happen inside Bpmn.readModelFromStream use try-with resources:

```suggestion
		try (InputStream resource = getClass().getClassLoader().getResource(bpmnFile).openStream())
		{
```

File: dsf-tools/dsf-tools-documentation-generator/src/main/java/org/highmed/dsf/tools/generator/DocumentationGenerator.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+	}
+
+	private String createDocumentation(Field field, List<String> pluginProcessNames)
+	{
+		Documentation documentation = field.getAnnotation(Documentation.class);
+		Value value = field.getAnnotation(Value.class);
+
+		String[] valueSplit = getValueDefaultArray(value.value());
+
+		String initialProperty = valueSplit[0];
```
Comment (hhund) [inline dsf-tools/dsf-tools-documentation-generator/src/main/java/org/highmed/dsf/tools/generator/DocumentationGenerator.java:143]: I have not looked closely: Is valueSplit always != null with length >= 1 ?

**Review by hhund — CHANGES_REQUESTED** [review]

File: dsf-tools/dsf-tools-documentation-generator/src/main/java/org/highmed/dsf/tools/generator/DocumentationGenerator.java (hunk 1 above)
Comment (wetret) [reply dsf-tools/dsf-tools-documentation-generator/src/main/java/org/highmed/dsf/tools/generator/DocumentationGenerator.java]: resolved with 6a6eb6f

File: dsf-tools/dsf-tools-documentation-generator/src/main/java/org/highmed/dsf/tools/generator/DocumentationGenerator.java (hunk 2 above)
Comment (wetret) [reply dsf-tools/dsf-tools-documentation-generator/src/main/java/org/highmed/dsf/tools/generator/DocumentationGenerator.java]: resolved with 6a6eb6f

**Review by hhund — APPROVED** [review]

**Comment by wetret** [comment]
Thanks for the additional improvements! I think the documentation generator is now ready for merging.

**Pull request merged**